examples/bri-3/src/bri/zkp/components/merkleTree/merkleTree.ts

```diff
@@ -0,0 +1,74 @@
+import { Tree } from "./tree"
+import { IMerkleTree} from "./merkleTree.interface";
+import { PreciseProofs } from "ew-precise-proofs-js";
+
+export class MerkleTree implements IMerkleTree {
+  tree: Tree;
+
+  constructor(document: any){
+    const merkleTree: Tree = this.createTree(document);
+
+    this.tree.leaves = merkleTree.leaves;
+    this.tree.root = merkleTree.root; 
+  }
+
+  private createTree(document: any): Tree {
+    try {
+      var tree: Tree;
+
+      const leaves: PreciseProofs.Leaf[] = this.createLeaves(document);
```

Comment:
what are the chances that leaves can be undefined here?

Reply:
in case leaves is undefined, leaves.map exception will be caught by the catch block bellow. In that case , this method should return something as Skos said (i.e undefined) and the caller then must check the return value before trying to assign values from it.